pkg/scaling/cache/scalers_cache.go

```diff
@@ -70,105 +69,53 @@ func (c *ScalersCache) GetPushScalers() []scalers.PushScaler {
 	return result
 }
 
-// GetMetricsForScaler returns metric value and latency for a scaler identified by the metric name
+// GetMetricSpecForScalingForScaler returns metrics spec for a scaler identified by the metric name
+func (c *ScalersCache) GetMetricSpecForScalingForScaler(ctx context.Context, index int) ([]v2.MetricSpec, error) {
```

Comment:
`GetMetricSpecForScalingForScaler` ? isn't it a bit redundant?

Reply:
yeah, I had the same concern 😄  But I am trying to be consistent with the other methods and also the method it is calling is named `GetMetricSpecForScaling` https://github.com/kedacore/keda/blob/6168d60d295067fa95701aa4f80697770b97394a/pkg/scalers/scaler.go#L50

Reply:
by consistent I mean: `GetMetricsAndActivity()` -> `GetMetricsAndActivityForScaler()`

Reply:
pff... you are right but I'm not still not sure about the naming :( where is the limit for appending things? As you prefer, for me this is not a blocker and I don't have any good suggestion more than `GetMetricSpecForScaling` (because it's the function that call us)

Reply:
I am not happy about this as well 😄 

let me merge this and I will try to think about a better name for the subsequent PRs